### PR TITLE
NF: represents counts as a class

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I don't like int[] when we know exactly that they are three elements. Furthermore, counts appear in enough place that it
makes sens to create a class for it. This makes for more readable code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java [hunk 1]
```diff
@@ -11,6 +13,10 @@
  * Represents the three counts shown in deck picker and reviewer. Semantically more meaningful than int[]
  */
 public class Counts {
+    public enum Column {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java:16]: It's a Queue not a Column really, isn't it? This feels like a mental hold-over from the int[0,1,2] previous vs naming it what it really represents

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java [hunk 2]
```diff
@@ -133,13 +133,13 @@ public void answerCard(@NonNull Card card, @Consts.BUTTON_TYPE int ease) {
...
 
     @Override
-    public int[] counts(@NonNull Card card) {
-        int[] counts = counts();
+    public Counts counts(@NonNull Card card) {
+        Counts counts = counts();
         int idx = countIdx(card);
         if (idx == 1) {
-            counts[1] += card.getLeft() / 1000;
+            counts.mLrn += card.getLeft() / 1000;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java:143]: it seems like all the Counts usage is encapsulated except this - here you just reach and set something, but the Counts object has some math that it does also. I'd prefer the queue members of Count to be actually encapsulated and not accessed here in this manner for consistency

**Review by mikehardy — CHANGES_REQUESTED** [review]
I like the idea - seems like a good typing improvement, but is partially encapsulated and has one naming thing - both of which I mentioned in specific comments

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java]: In my head, I was thinking about the column of number in the deck picker. It counts the number of elements in the queue. But I honestly don't care about the name. I find `.getNew()` far more than `[0]`, then whether it's a queue or a column matter little

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java]: Corrected.

**Review by mikehardy — APPROVED** [review]
LGTM - thanks!

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Counts.java]: I agree it is not the most important thing, but access to these can be completely unordered so do like queue better. Thanks

**Comment by mikehardy** [comment]
Has legit unit test failures

https://travis-ci.org/github/ankidroid/Anki-Android/jobs/728813676

**Comment by mikehardy** [comment]
Still not working for some reason and took a conflict now :sweat_smile:

**Comment by Arthur-Milchior** [comment]
Corrected. I inversed elements 1 and 2 of an array... which is kind of funny because it was exactly the reason why I wanted this change, to ensure that this kind of error does not occur ever

**Review by mikehardy — APPROVED** [review]

**Pull request merged**